src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1217,6 +1221,25 @@ public static void applySslContext()
         }
     }
 
+    public static void applyCryptoProvider()
+    {
+        try
+        {
+            if (conf.crypto_provider == null)
+                conf.crypto_provider = new ParameterizedClass("org.apache.cassandra.security.DefaultCryptoProvider", null);
+
+            Class<?> cryptoProviderClass = Class.forName(conf.crypto_provider.class_name);
+            cryptoProvider = (ICryptoProvider)cryptoProviderClass.getConstructor(Map.class).newInstance(conf.crypto_provider.parameters);
+
+            cryptoProvider.installProvider();
+        }
+        catch(Exception e)
+        {
+            throw new ConfigurationException("Failed to initialize crypto Provider.", e);
```

Comment:
It would be nice to say what was the class (from class_name) of the provider we failed to initialize.